### ISSUE TITLE
Pods not getting evicted even after node is notready when using taint based eviction.

### ISSUE THREAD

**Issue by aveshagarwal** [kubernetes/kubernetes#42733]
I am just starting a local cluster by using hack/local-up-cluster.sh and the cluster has taint controller (--enable-taint-manager) and taint based evictions (--use-taint-based-evictions) enabled. I create a very simple pod on this cluster. Now when I kill kublet, I see that the node, after some time, goes into NotReday state.

```
# ./_output/bin/kubectl get nodes
NAME        STATUS     AGE       VERSION
127.0.0.1   NotReady   10h       v1.7.0-alpha.0.927+7593a9f55548af-dirty
```
After waiting for some time (infact lot of time) I dont see the node is getting any NotReady or Unreachable taint

```
#  ./_output/bin/kubectl describe nodes
Name:			127.0.0.1
Role:			
Labels:			alpha.kubernetes.io/fluentd-ds-ready=true
			beta.kubernetes.io/arch=amd64
			beta.kubernetes.io/os=linux
			kubernetes.io/hostname=127.0.0.1
Annotations:		node.alpha.kubernetes.io/ttl=0
			volumes.kubernetes.io/controller-managed-attach-detach=true
Taints:			<none>
```

The pod stays running even after several hours:

```
# ./_output/bin/kubectl get pods
NAME            READY     STATUS    RESTARTS   AGE
dapi-test-pod   1/1       Running   0          10h
```

And here are the controller-manager logs:

```
I0308 00:42:56.119517   21558 event.go:217] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"127.0.0.1", UID:"6b54f6e6-03c1-11e7-a627-525400b5933e", APIVersion:"", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'NodeNotReady' Node 127.0.0.1 status is now: NodeNotReady
I0308 00:42:56.120498   21558 controller_utils.go:219] Updating ready status of pod dapi-test-pod to false
I0308 00:42:56.121862   21558 nodecontroller.go:765] NodeController detected that all Nodes are not-Ready. Entering master disruption mode.
I0308 00:43:01.122458   21558 nodecontroller.go:656] Node 127.0.0.1 is unresponsive as of 2017-03-08 00:43:01.122443088 -0500 EST. Adding it to the Taint queue.
W0308 00:57:01.679296   21558 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: etcdserver: mvcc: required revision has been compacted
I0308 00:57:02.679585   21558 reflector.go:236] Listing and watching <nil> from k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190
W0308 01:05:32.783028   21558 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: etcdserver: mvcc: required revision has been compacted
I0308 01:05:33.783349   21558 reflector.go:236] Listing and watching <nil> from k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190
W0308 01:20:24.863511   21558 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: etcdserver: mvcc: required revision has been compacted
I0308 01:20:25.863805   21558 reflector.go:236] Listing and watching <nil> from k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190
W0308 01:34:50.870730   21558 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: etcdserver: mvcc: required revision has been compacted
```

The last taint related msg I see is from the above logs:
```
I0308 00:43:01.122458   21558 nodecontroller.go:656] Node 127.0.0.1 is unresponsive as of 2017-03-08 00:43:01.122443088 -0500 EST. Adding it to the Taint queue.
```

I am wondering if something is wrong with my cluster or is it really an issue with the code? Though I am debugging in parallel but just thought to post here to see of someone has any suggestion.

@kubernetes/sig-scheduling-bugs

**Comment by gmarek** [comment]
`local-up-cluster` brings up a single Node cluster, which is the reason why you don't see any evictions. Everything's working as intended. Important line is this one:
```
I0308 00:42:56.121862   21558 nodecontroller.go:765] NodeController detected that all Nodes are not-Ready. Entering master disruption mode.
```

Around August last year we introduced a protection against master machine network isolation, which prevents any evictions from happening if master can't see any healthy Node. It then assumes that it's not a problem with Nodes, but with itself and just don't do anything.

I thought for a sec about special-casing single-node clusters, but I didn't see a point in that (evicted pods won't be scheduled anywhere else, as there's no anywhere else).

So to sum it up - you need to have at least two machine cluster to observe eviction.

**Comment by aveshagarwal** [comment]
@gmarek Now I tried it in a 1 master and 2-node cluster where master and 1 node is on one VM and another node is on another VM.  

The only different I see that the node that goes down is now getting NoExecute taint assigned. However pod eviction is still not happening:

```
# _output/bin/kubectl get nodes
NAME              STATUS     AGE       VERSION
192.168.124.120   Ready      41d       v1.7.0-alpha.0.992+39fde0708e6738
192.168.124.214   NotReady   41d       v1.7.0-alpha.0.990+c3d7ee11d176d9-dirty
```

And see that pod still remain assigned to 192.168.124.214 even after almost 13 mins:

```
# _output/bin/kubectl get pods -o wide
NAME      READY     STATUS    RESTARTS   AGE       IP           NODE
busybox   1/1       Running   0          13m       18.16.71.2   192.168.124.214
```

And I see that the node 192.168.124.214 is getting unreachable NoEexcute taint:

```
Name:			192.168.124.214
Role:			
Labels:			alpha.kubernetes.io/fluentd-ds-ready=true
			beta.kubernetes.io/arch=amd64
			beta.kubernetes.io/os=linux
			kubernetes.io/hostname=192.168.124.214
Annotations:		node.alpha.kubernetes.io/ttl=0
			volumes.kubernetes.io/controller-managed-attach-detach=true
Taints:			node.alpha.kubernetes.io/unreachable:NoExecute
CreationTimestamp:	Thu, 26 Jan 2017 11:05:23 -0500
Phase:			
Conditions:
  Type			Status		LastHeartbeatTime			LastTransitionTime			Reason			Message
  ----			------		-----------------			------------------			------			-------
  OutOfDisk 		Unknown 	Wed, 08 Mar 2017 13:44:56 -0500 	Wed, 08 Mar 2017 13:45:39 -0500 	NodeStatusUnknown 	Kubelet stopped posting node status.
  MemoryPressure 	Unknown 	Wed, 08 Mar 2017 13:44:56 -0500 	Wed, 08 Mar 2017 13:45:39 -0500 	NodeStatusUnknown 	Kubelet stopped posting node status.
  DiskPressure 		Unknown 	Wed, 08 Mar 2017 13:44:56 -0500 	Wed, 08 Mar 2017 13:45:39 -0500 	NodeStatusUnknown 	Kubelet stopped posting node status.
  Ready 		Unknown 	Wed, 08 Mar 2017 13:44:56 -0500 	Wed, 08 Mar 2017 13:45:39 -0500 	NodeStatusUnknown 	Kubelet stopped posting node status.
Addresses:		192.168.124.214,192.168.124.214,192.168.124.214
Capacity:
 alpha.kubernetes.io/nvidia-gpu:	0
 cpu:					4
 memory:				8175612Ki
 pods:					110
Allocatable:
 alpha.kubernetes.io/nvidia-gpu:	0
 cpu:					4
 memory:				8073212Ki
 pods:					110
System Info:
 Machine ID:			7433817332b74356a2fdd05f6644c13b
 System UUID:			299C2387-6623-4E34-A940-220CC9F00EC4
 Boot ID:			d72c023c-2d83-41ed-ad9b-bcfc3403986c
 Kernel Version:		4.9.13-200.fc25.x86_64
 OS Image:			Fedora 25 (Workstation Edition)
 Operating System:		linux
 Architecture:			amd64
 Container Runtime Version:	docker://1.12.6
 Kubelet Version:		v1.7.0-alpha.0.990+c3d7ee11d176d9-dirty
 Kube-Proxy Version:		v1.7.0-alpha.0.990+c3d7ee11d176d9-dirty
ExternalID:			192.168.124.214
Non-terminated Pods:		(1 in total)
  Namespace			Name		CPU Requests	CPU Limits	Memory Requests	Memory Limits
  ---------			----		------------	----------	---------------	-------------
  default			busybox		0 (0%)		0 (0%)		0 (0%)		0 (0%)
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  CPU Requests	CPU Limits	Memory Requests	Memory Limits
  ------------	----------	---------------	-------------
  0 (0%)	0 (0%)		0 (0%)		0 (0%)
Events:
  FirstSeen	LastSeen	Count	From				SubObjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  41m		41m		1	kubelet, 192.168.124.214			Normal		Starting		Starting kubelet.
  41m		41m		1	kubelet, 192.168.124.214			Warning		ImageGCFailed		unable to find data for container /
  41m		41m		2	kubelet, 192.168.124.214			Normal		NodeHasSufficientDisk	Node 192.168.124.214 status is now: NodeHasSufficientDisk
  41m		41m		1	kubelet, 192.168.124.214			Normal		NodeHasSufficientMemory	Node 192.168.124.214 status is now: NodeHasSufficientMemory
  41m		41m		1	kubelet, 192.168.124.214			Normal		NodeHasNoDiskPressure	Node 192.168.124.214 status is now: NodeHasNoDiskPressure
  41m		41m		1	kubelet, 192.168.124.214			Warning		Rebooted		Node 192.168.124.214 has been rebooted, boot id: d72c023c-2d83-41ed-ad9b-bcfc3403986c
  41m		41m		1	kubelet, 192.168.124.214			Normal		NodeNotReady		Node 192.168.124.214 status is now: NodeNotReady
  41m		41m		1	kubelet, 192.168.124.214			Normal		NodeReady		Node 192.168.124.214 status is now: NodeReady
```

**Comment by aveshagarwal** [comment]
@gmarek  its been alsmot 29 mins now.  and the node is still NotReady. 

```
# _output/bin/kubectl describe pods
Name:		busybox
Namespace:	default
Node:		192.168.124.214/192.168.124.214
Start Time:	Wed, 08 Mar 2017 13:43:57 -0500
Labels:		service=S1
Annotations:	<none>
Status:		Running
IP:		18.16.71.2
Controllers:	<none>
Containers:
  busybox:
    Container ID:	docker://fb949a09c0be292d90cc47eaf365db07c29c897aa17d3a6233691186cd172b05
    Image:		busybox
    Image ID:		docker-pullable://docker.io/busybox@sha256:817a12c32a39bbe394944ba49de563e085f1d3c5266eb8e9723256bc4448680e
    Port:		
    Command:
      sleep
      999999
    State:		Running
      Started:		Wed, 08 Mar 2017 13:44:01 -0500
    Ready:		True
    Restart Count:	0
    Environment:	<none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-gckq7 (ro)
Conditions:
  Type		Status
  Initialized 	True 
  Ready 	False 
  PodScheduled 	True 
Volumes:
  default-token-gckq7:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-gckq7
    Optional:	false
QoS Class:	BestEffort
Node-Selectors:	<none>
Tolerations:	<none>
Events:
  FirstSeen	LastSeen	Count	From				SubObjectPath			Type		Reason		Message
  ---------	--------	-----	----				-------------			--------	------		-------
  29m		29m		1	default-scheduler						Normal		Scheduled	Successfully assigned busybox to 192.168.124.214
  29m		29m		1	kubelet, 192.168.124.214	spec.containers{busybox}	Normal		Pulling		pulling image "busybox"
  29m		29m		1	kubelet, 192.168.124.214	spec.containers{busybox}	Normal		Pulled		Successfully pulled image "busybox"
  29m		29m		1	kubelet, 192.168.124.214	spec.containers{busybox}	Normal		Created		Created container with id fb949a09c0be292d90cc47eaf365db07c29c897aa17d3a6233691186cd172b05
  29m		29m		1	kubelet, 192.168.124.214	spec.containers{busybox}	Normal		Started		Started container with id fb949a09c0be292d90cc47eaf365db07c29c897aa17d3a6233691186cd172b05
```

**Comment by gmarek** [comment]
OK, then there's something wrong. Taint Controller didn't even tried to do anything with this Pod. Can you please get logs from `kube-controller-manager` that references this Pod and Node?

**Comment by aveshagarwal** [comment]
```
Mar 08 12:54:33 fedora25 systemd[1]: Started Kubernetes Controller Manager.
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.060379   32146 feature_gate.go:144] feature gates: map[ExperimentalCriticalPodAnnotation:true Accelerators:true TaintBasedEvictions:true AffinityInAnnota
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.062299   32146 leaderelection.go:179] attempting to acquire leader lease...
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.065762   32146 leaderelection.go:189] successfully acquired lease kube-system/kube-controller-manager
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.067386   32146 event.go:217] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"kube-controller-manager", UID:"bd5b2e80-e40a-11e6-
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.080645   32146 controllermanager.go:437] Started "deployment"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.080951   32146 controllermanager.go:437] Started "disruption"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.081152   32146 controllermanager.go:437] Started "endpoint"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.081368   32146 controllermanager.go:437] Started "daemonset"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.081592   32146 deployment_controller.go:147] Starting deployment controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.081642   32146 disruption.go:269] Starting disruption controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.081689   32146 daemoncontroller.go:199] Starting Daemon Sets controller manager
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.382854   32146 controllermanager.go:437] Started "namespace"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.383003   32146 namespace_controller.go:189] Starting the NamespaceController
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.390294   32146 controllermanager.go:437] Started "garbagecollector"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.390356   32146 garbagecollector.go:111] Garbage Collector: Initializing
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.399660   32146 controllermanager.go:437] Started "replicaset"
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.399681   32146 controllermanager.go:434] Skipping "cronjob"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.399757   32146 replica_set.go:155] Starting ReplicaSet controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.399810   32146 certificates.go:38] Failed to start certificate controller: open /etc/kubernetes/ca/ca.pem: no such file or directory
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.399823   32146 controllermanager.go:434] Skipping "certificatesigningrequests"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.399995   32146 controllermanager.go:437] Started "ttl"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.400137   32146 ttlcontroller.go:117] Starting TTL controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.400271   32146 controllermanager.go:437] Started "replicationcontroller"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.400357   32146 replication_controller.go:150] Starting RC Manager
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.400958   32146 controllermanager.go:437] Started "podgc"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.404809   32146 controllermanager.go:437] Started "job"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.405280   32146 controllermanager.go:437] Started "horizontalpodautoscaling"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.405396   32146 horizontal.go:142] Starting HPA Controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.405582   32146 controllermanager.go:437] Started "statefuleset"
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.405594   32146 controllermanager.go:421] "bootstrapsigner" is disabled
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.405601   32146 controllermanager.go:421] "tokencleaner" is disabled
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.405640   32146 stateful_set.go:144] Starting statefulset controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406048   32146 controllermanager.go:437] Started "resourcequota"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406108   32146 resource_quota_controller.go:240] Starting resource quota controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.406285   32146 util.go:45] Metric for serviceaccount_controller already registered
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406309   32146 controllermanager.go:437] Started "serviceaccount"
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406315   32146 plugins.go:101] No cloud provider specified.
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.406323   32146 controllermanager.go:449] Unsuccessful parsing of cluster CIDR : invalid CIDR address:
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.406331   32146 controllermanager.go:453] Unsuccessful parsing of service CIDR : invalid CIDR address:
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406374   32146 serviceaccounts_controller.go:122] Starting ServiceAccount controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406495   32146 nodecontroller.go:219] Sending events to api server.
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406551   32146 nodecontroller.go:268] NodeController is using taint based evictions.
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406576   32146 taint_controller.go:198] Sending events to api server.
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.406950   32146 controllermanager.go:494] Failed to start service controller: WARNING: no cloud provider provided, services of type LoadBalancer will fail
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.406962   32146 controllermanager.go:506] Unsuccessful parsing of cluster CIDR : invalid CIDR address:
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.406969   32146 controllermanager.go:519] Will not configure cloud provider routes for allocate-node-cidrs: false, configure-cloud-routes: true.
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.411635   32146 attach_detach_controller.go:223] Starting Attach Detach Controller
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.415393   32146 actual_state_of_world.go:461] Failed to set statusUpdateNeeded to needed true because nodeName="192.168.124.120"  does not exist
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.415404   32146 actual_state_of_world.go:475] Failed to update statusUpdateNeeded field in actual state of world: Failed to set statusUpdateNeeded to need
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.415412   32146 actual_state_of_world.go:461] Failed to set statusUpdateNeeded to needed true because nodeName="192.168.124.214"  does not exist
Mar 08 12:54:34 fedora25 hyperkube[32146]: E0308 12:54:34.415415   32146 actual_state_of_world.go:475] Failed to update statusUpdateNeeded field in actual state of world: Failed to set statusUpdateNeeded to need
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.481889   32146 disruption.go:277] Sending events to api server.
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.490646   32146 garbagecollector.go:116] Garbage Collector: All resource monitors have synced. Proceeding to collect garbage
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.506852   32146 taint_controller.go:221] Starting NoExecuteTaintManager
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.506856   32146 nodecontroller.go:581] Initializing eviction metric for zone:
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.506952   32146 event.go:217] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"192.168.124.120", UID:"65ae96ed-e3e1-11e6-aa09-5254009d44b2", APIV
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.507202   32146 event.go:217] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"192.168.124.214", UID:"3e811e9a-e3e1-11e6-aa09-5254009d44b2", APIV
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.507178   32146 nodecontroller.go:904] Missing timestamp for Node 192.168.124.120. Assuming now as a timestamp.
Mar 08 12:54:34 fedora25 hyperkube[32146]: W0308 12:54:34.507239   32146 nodecontroller.go:904] Missing timestamp for Node 192.168.124.214. Assuming now as a timestamp.
Mar 08 12:54:34 fedora25 hyperkube[32146]: I0308 12:54:34.507256   32146 nodecontroller.go:770] NodeController detected that all Nodes are not-Ready. Entering master disruption mode.
Mar 08 13:14:50 fedora25 hyperkube[32146]: W0308 13:14:50.406905   32146 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: 401: The event in req
Mar 08 13:16:34 fedora25 hyperkube[32146]: I0308 13:16:34.604861   32146 nodecontroller.go:797] NodeController detected that some Nodes are Ready. Exiting master disruption mode.
Mar 08 13:36:27 fedora25 hyperkube[32146]: W0308 13:36:27.413421   32146 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: 401: The event in req
Mar 08 13:45:39 fedora25 hyperkube[32146]: I0308 13:45:39.752365   32146 event.go:217] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"192.168.124.214", UID:"3e811e9a-e3e1-11e6-aa09-5254009d44b2", APIV
Mar 08 13:50:46 fedora25 hyperkube[32146]: W0308 13:50:46.417997   32146 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: 401: The event in req
Mar 08 14:08:18 fedora25 hyperkube[32146]: W0308 14:08:18.422693   32146 reflector.go:323] k8s.io/kubernetes/pkg/controller/garbagecollector/graph_builder.go:190: watch of <nil> ended with: 401: The event in r
```

**Comment by aveshagarwal** [comment]
@gmarek One issue I see that I dont know why at one point controller-manager detected all nodes not ready and entered master disruption mode, and a few moments later, it again sees that some nodes are ready but still does not start eviction.

**Comment by gmarek** [comment]
When the cluster starts and first Node joins it's normal that "all Nodes are not Ready" (as it takes some time for Nodes to become ready), and later on other Nodes join, and they become ready. I wouldn't worry about that. We need to see what happened on the Node that became not ready and with the busybox pod that was running on it.

**Comment by aveshagarwal** [comment]
@gmarek are you looking for any more info/data from me?

**Comment by gmarek** [comment]
Yup - logs from controller manager, that mentions busybox pod.

**Comment by aveshagarwal** [comment]
@gmarek those are only logs from controller manager I have.

**Comment by aveshagarwal** [comment]
@gmarek is https://github.com/kubernetes/kubernetes/issues/42733#issuecomment-285140872 lacking something?

**Comment by gmarek** [comment]
Are you running with `--v=2`?

**Comment by gmarek** [comment]
It lacks a lot of things... Normally controller-manager generates a large amount of logs every minute. If it's silent for an hour, then either your log level is too low, or something's really wrong.

**Comment by aveshagarwal** [comment]
@gmarek --v=0

**Comment by gmarek** [comment]
OK, with `--v=0` we can't tell anything:). Please redo the experiment with (at least) `--v=2`.

**Comment by aveshagarwal** [comment]
ok sure will post here after some time.

**Comment by gyliu513** [comment]
>using hack/local-up-cluster.sh and the cluster has taint controller (--enable-taint-manager) and taint based evictions (--use-taint-based-evictions) enabled

@aveshagarwal Just a question here, for `enable-taint-manager`, I found that the default value is `True`; Regarding to `use-taint-based-evictions`, I did not found any place in code referring this parameter and do not know which component should use this parameter.

This is the only place in code listing the `use-taint-based-evictions` https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/well_known_labels.go#L37-L46 , can you please show me how you enable this parameter? Thanks!

**Comment by k82cn** [comment]
moved by  https://github.com/kubernetes/kubernetes/commit/48d784272eccc2a59505ed9ece590d04143fec7f

**Comment by gyliu513** [comment]
Thanks @k82cn , good to know, created a PR to update https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/well_known_labels.go#L37-L46 at https://github.com/kubernetes/kubernetes/pull/42786

**Comment by aveshagarwal** [comment]
@gmarek I am still seeing the issue and this time I ran controller-manager with --v=5. Here is the link for logs: https://paste.fedoraproject.org/paste/qrpwjZOLVJPFtci3yCr4I15M1UNdIGYhyRLivL9gydE=

**Comment by aveshagarwal** [comment]
@gmarek also could you reopen this issue?
